ms.go

```diff
@@ -0,0 +1,140 @@
+package ms
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/coreos/go-oidc/v3/oidc"
+	"os"
+	"strings"
+)
+
+const rolesSeparator = "."
+const rolesKey = "roles"
+
+// Service defines basic parameters of a microservice for a client.
+type Service struct {
+	Host string
+	Port string
+}
+
+// BaseServiceServer contains a port on which the server listens for GRPC requests and provides JWT token verification.
+type BaseServiceServer struct {
```

Comment:
Aren't interface suitable only for method specification? 
Use case of `BaseServiceServer`, `Claims` and `GetRequiredEnv`:
We want to implement some MS that has additional parameter which will be provided via env variable `EXTRA_FIELD`.

Firstly we define a struct for our MS that promotes:
```go
type MyServer struct {
	*BaseServiceServer
	additionalField string
}
```

It will be initialized in the following way:
```go
var s MyServer
s.BaseServiceServer, _ = CreateBaseServiceServer()
s.additionalField, _ = GetRequiredEnv("EXTRA_FIELD")
```

Implementation of `someGRPCFunc` that requires authentication and authorization:
```go
func (server MyServer) someGRPCFunc(ctx context.Context, token string) {
	// check authentication
	claims, err := server.VerifyToken(ctx, token)
	if err != nil {
		// some error
	}

	// check authorization
	authorized := claims.HasRole("doctor")
	if !authorized {
		// some error
	}
}
```